File: cpp/gms/gms-filters-provider/src/descriptions/LinearFIRFilterDescription.cpp
#include "LinearFIRFilterDescription.hh"

LinearFIRFilterDescription::LinearFIRFilterDescription(FIRFilterParameters parameters,
                                                       bool causal,
                                                       std::string comments,
                                                       double highFrequency,
                                                       double lowFrequency,
                                                       int passBandType,
                                                       int filterDesignModel,
                                                       int order,
                                                       bool zeroPhase) : parameters(parameters),
                                                                        BaseLinearFilterDescription(causal,
                                                                                                    comments,
                                                                                                    highFrequency,
                                                                                                    lowFrequency,
                                                                                                    passBandType,
                                                                                                    filterDesignModel,
                                                                                                    order,
                                                                                                    zeroPhase){};

File: typescript/user-interface/packages/ui-wasm/src/ts/gms-filters-provider/gms-filters/types/linear-fir-filter-description.ts
import type { WasmFilterComputationType } from './filter-computation-type';
import type { WasmFIRFilterParameters } from './fir-filter-parameters';
import type { WasmLinearFilterDescription } from './linear-filter-description';

export interface WasmLinearFIRFilterDescriptionModule {
  new (
    parameters: WasmFIRFilterParameters,
    causal: boolean,
    comments: string,
    cutoffHigh: number,
    cutoffLow: number,
    filterBandType: number,
    filterDesignModel: number,
    filterOrder: number,
    zeroPhase: boolean
  ): WasmLinearFIRFilterDescription;
}
export interface WasmLinearFIRFilterDescription extends WasmLinearFilterDescription {
  $$: {
    ptr: number;
  };
  delete: () => void;
  filterType: WasmFilterComputationType.FIR;
  firFilterParameters: WasmFIRFilterParameters;
}
